File: 백준/Silver/2839. 설탕 배달/설탕 배달.py
# N킬로그램 배달
# 봉지 -> 3kg, 5kg
# 최대한 적은 봉지
import sys
N = int(sys.stdin.readline())

res = N % 5
answer = 0
while N >= 0 :
    if N % 5 == 0 :
        answer += N // 5
        print(answer)
        break
    N -= 3
    answer += 1
else :
    answer = -1
    print(answer)

File: 백준/Silver/2839. 설탕 배달/설탕 배달.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        int N = Integer.parseInt(br.readLine());

        int res = N % 5;
        int answer = 0;

        while (N >= 0) {
            if (N % 5 == 0) {
                answer += N / 5;
                System.out.println(answer);
                break;
            }
            N -= 3;
            answer += 1;
        }

        if (N < 0) {
            answer = -1;
            System.out.println(answer);
        }
    }
}